docs/lesson-plans/week-1.md

```diff
@@ -0,0 +1,649 @@
+
+>Transpilation is when instead of 'compiling' code, the code gets translated to a different language.
+>In this case, Typescript is being converted to Javascript.
+
+>Linting is when the code is checked for formatting and against common style guides.
+
+Common commands include:
+- `npm install`: install dependencies for a project
+- `npm install -g`: install a global dependency (typically needs `sudo`)
+- `npm start`: starts a server
+- `npm test`: runs tests
+- `npm run <command>`: runs a command in the package.json scripts section
+- `npm run build`: a React command (can be supplied with other things though) to prepare your
+project for a production environment (compress everything and set up resources).
+
+#### Maven
+
+Maven is very similar to npm in a lot of ways. It is one of the popular project build
+and dependency managers for Java. Similarly to npm, it allows you to import plugins 
+(lets you run programs outside of your own program) to format code and adjust how
+your project is built. It also allows you to create these sort-of sub-projects, which 
+provide modularity and flexibility within your code. 
+
+If you take a look though the project, you might notice multiple pom.xml files.
+One main pom.xml in the project root, and three more (one in the subdirectories api/, persist/,
+and /service). The way we set up this project is so that each of the subdirectories are
+modules, and therefore sub-projects of the main backend project. 
+
+>Modules are groups of code that are compiled separately of each other, however, they can be used 
+>as depedndencies of other modules. If you take a look at the service pom.xml, you'll notice that
+>there are dependencies on the persist and api modules.
+
+Each command in Maven is called a 'goal', and goals can make up parts of other goals. You 
+can also run goals individually or together. For example, `mvn`, `mvn clean install`, 
+and `mvn clean spotless:apply install` all have the same result.
+
+Common commands include:
+- `mvn`: the default 'goal'. This runs 2 important tasks in the Maven 
+['lifecycle'](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html);
+`mvn clean` and `mvn install`
+- `mvn clean`: deletes built artifacts and compiled code so that it can be rebuilt later
+- `mvn install`: compiles code so that it can be run later on (and also applies the 
+`mvn spotless:apply` command)
+- `mvn spotless:apply`: performs code formatting to nicely and uniformly clean up your code
+
+A flag which you may want to use every once in a while when you're having trouble compiling a
+code because of a test, but you want to have it compiled so that you can run the test
+using the IntelliJ debugger is the `-DskipTests` flag. You can append it to the end
+of the `install` goal above, and it will skip tests so that you can successfully compile
+(if you're not able to compile because of a test). Remember though, this is not a command
```

Comment:
maybe remove the sentence in the () because it was mentioned already

Reply:
I took it out.